switchbot/adv_parsers/keypad.py

```diff
@@ -0,0 +1,24 @@
+"""Keypad parser."""
+
+from __future__ import annotations
+
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def process_wokeypad(
+    data: bytes | None, mfr_data: bytes | None
+) -> dict[str, bool | int | None] | {}:
+    """Process woKeypad services data."""
+    if data is None and mfr_data is None:
+        return {}
+
+    if data is None:
+        return {"battery": None, "attemptState": None}
+
+    _LOGGER.debug("mfr_data: %s", mfr_data.hex())
```

Comment:
`mfr_data` can be `None` here which would raise when trying to call `.hex()` on it